src/libsync/capabilities.cpp

```diff
@@ -63,9 +65,28 @@ bool Capabilities::shareResharing() const
     return _capabilities["files_sharing"].toMap()["resharing"].toBool();
 }
 
-QStringList Capabilities::supportedChecksumTypes() const
+QList<QByteArray> Capabilities::supportedChecksumTypesRaw() const
 {
-    return QStringList();
+    return QList<QByteArray>();
+}
+
+QList<QByteArray> Capabilities::supportedChecksumTypes() const
+{
+    auto list = supportedChecksumTypesRaw();
+    QByteArray cfgType = ConfigFile().transmissionChecksum().toLatin1();
+    if (!cfgType.isEmpty()) {
+        list.prepend(cfgType);
+    }
+    return list;
+}
+
+QByteArray Capabilities::preferredChecksumType() const
+{
+    auto list = supportedChecksumTypes();
+    if (list.isEmpty()) {
+        return QByteArray();
+    }
+    return list.first();
 }
 
 }
```

Comment:
I assume the function `supportedChecksumTypesRaw()` is supposed to fetch the supported checksums from the server capabilities, but that is not yet implemented, right? Why not name it like that? 'raw' is pretty generic imo.

Also, if we have a list of supported ones from the server, the prepend can lead to duplicate entries, right?